### PR TITLE
Feature/addbinarydatabson

### PR DESCRIPTION
**Pull request by dehann**
Hi @felipenoris , thanks for this package -- please find read and write binary data of `BSON_TYPE_BINARY`.  If you are happy with these changes, it would be great if you could also tag a new release with JuliaLang/METADATA.jl too please?

PS, I was having a little trouble with Julia 1.0 and Mongoc, but see the Travis tests are passing.  The package should currently work on Julia 1.0, right?

### CODE REVIEW

**Comment by dehann** [comment]
helps with JuliaLang/METADATA.jl#18786

**Comment by dehann** [comment]
Having trouble with Julia 1.0 not recognizing `Void` on Mongoc.jl lines 6 and 7, even though 1.0 should not execute those lines.

**Comment by dehann** [comment]
I tried looking at Compat.jl, but still had the same issue.  I think something has changed with regard to precompiling in Julia 1.0 somehow (really strange though).  The `if` statement at the start of Mongoc.jl does not seem to be working the same it did 8 days ago on Travis.  Do you perhaps know how to get around this Void issue  - and maybe help fix it please?

File: src/c_api.jl [hunk 1]
```diff
@@ -164,6 +168,12 @@ function bson_iter_code(iter_ref::Ref{BSONIter})
     ccall((:bson_iter_code, libbson), Cstring, (Ref{BSONIter}, Ptr{UInt32}), iter_ref, C_NULL)
 end
 
+function bson_iter_binary(iter_ref::Ref{BSONIter}, lengthPtr::Array{UInt32}, dataPtr::Array{Ptr{UInt8}})
+    bsonsubtype = BSON_SUBTYPE_BINARY
+    ccall((:bson_iter_binary, libbson), Ptr{Void}, (Ref{BSONIter}, Ref{BSONSubType}, Ptr{UInt32}, Ptr{Ptr{UInt8}}),iter_ref, bsonsubtype, lengthPtr, dataPtr)
```
Comment (felipenoris) [inline src/c_api.jl]: Looks like you should replace `Ptr{Void}` with `Cvoid`.

**Comment by felipenoris** [comment]
Please, add some tests to the new features.

File: src/c_api.jl (hunk 1 above)
Comment (dehann) [reply src/c_api.jl]: that worked thanks! Will add binary data tests soon.

**Comment by dehann** [comment]
Hi @felipenoris , all clear -- tests added too.  Thanks for the `Cvoid` suggestion.  Maybe you could tag a release for METADATA.jl if you don't mind please.

**Comment by felipenoris** [comment]
Nice! Could you please squash your commits? See https://www.devroom.io/2011/07/05/git-squash-your-latests-commits-into-one/ .

**Comment by dehann** [comment]
Could you not just use the "squash and merge" feature in Github's merge command for this PR?

**Comment by dehann** [comment]
Great, thanks!

**Pull request merged**